test/ssl_test_ctx.c

```diff
@@ -480,6 +480,39 @@ __owur static int parse_expected_client_cert_type(SSL_TEST_CTX *test_ctx,
                                    value);
 }
 
+/*************************/
+/* Expected signing hash */
+/*************************/
+
+__owur static int parse_expected_sign_hash(int *ptype, const char *value)
+{
+    int nid;
+
+    if (value == NULL)
+        return 0;
+    nid = OBJ_sn2nid(value);
+    if (nid == NID_undef)
+        nid = OBJ_ln2nid(value);
+    if (nid == NID_undef)
```

Comment:
Start be rewriting the code:
```
 nid = OBJ_sn2nid(value);
 if (nid == NID_undef) {
        nid = OBJ_ln2nid(value);
       if (nid == NID_undef)
          return 0;
}
```
Right?

The code is looking up things under *two* names and if both fail, then it returns.
So it should be ```&&```